atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/RandomWorkflows.java

```diff
@@ -0,0 +1,118 @@
+import com.palantir.atlasdb.workload.transaction.ImmutableDeleteTransactionAction;
+import com.palantir.atlasdb.workload.transaction.ImmutableReadTransactionAction;
+import com.palantir.atlasdb.workload.transaction.ImmutableWriteTransactionAction;
+import com.palantir.atlasdb.workload.transaction.ReadTransactionAction;
+import com.palantir.atlasdb.workload.transaction.TransactionAction;
+import com.palantir.atlasdb.workload.transaction.WriteTransactionAction;
+import com.palantir.atlasdb.workload.transaction.witnessed.WitnessedTransaction;
+import java.security.SecureRandom;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import org.jetbrains.annotations.VisibleForTesting;
+
+public final class RandomWorkflows {
+    private static final SecureRandom RANDOM = DefaultNativeSamplingSecureRandomFactory.INSTANCE.create();
+    private static final Integer COLUMN = 0;
+
+    private RandomWorkflows() {
+        // static factory
+    }
+
+    public static Workflow create(
+            TransactionStore store,
+            RandomWorkflowConfiguration randomWorkflowConfig,
+            ListeningExecutorService executionExecutor) {
+        return create(store, randomWorkflowConfig, executionExecutor, RANDOM);
+    }
+
+    @VisibleForTesting
+    static Workflow create(
+            TransactionStore store,
+            RandomWorkflowConfiguration randomWorkflowConfig,
+            ListeningExecutorService executionExecutor,
+            SecureRandom random) {
+        RandomWorkflowTask task = new RandomWorkflowTask(randomWorkflowConfig, random);
+        return DefaultWorkflow.create(
+                store, (txnStore, _index) -> task.run(txnStore), randomWorkflowConfig, executionExecutor);
+    }
+
+    private static final class RandomWorkflowTask {
+
+        private final RandomWorkflowConfiguration workflowConfiguration;
+        private final SecureRandom random;
+
+        public RandomWorkflowTask(RandomWorkflowConfiguration workflowConfiguration, SecureRandom random) {
+            this.workflowConfiguration = workflowConfiguration;
+            this.random = random;
+        }
+
+        public Optional<WitnessedTransaction> run(TransactionStore store) {
+            workflowConfiguration.transactionRateLimiter().acquire();
+            List<TransactionAction> actions = Stream.of(
+                            generateReadActions(), generateWriteActions(), generateDeleteActions())
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toList());
+            Collections.shuffle(actions, random);
+            return store.readWrite(actions);
+        }
+
+        private List<ReadTransactionAction> generateReadActions() {
+            return IntStream.rangeClosed(0, random.nextInt(workflowConfiguration.maxReads()))
```

Comment:
I did think about rangeClosed, though note that this means we will have _at least one_ of each operation type. If that's the intention that's fine, but wanted to make sure!

Reply:
ah shoot

Reply:
yeah we definitely do not want that